### ISSUE TITLE
Takeover=nonempty treats template text as a non-empty box

### ISSUE THREAD

**Issue by BlueDrink9** [glacambre/firenvim#470]
The comments section for sites like gitlab have elements for their textboxes that contain placeholder text. These seem to trigger Firenvim when takeover=nonempty, although from the user perspective these boxes are empty.

An example element:

```html
<textarea class="note-textarea js-note-text js-gfm-input js-autosize markdown-area" placeholder="Write a comment or drag your files here…" dir="auto" data-supports-quick-actions="true" data-supports-autocomplete="true" data-qa-selector="" name="note[note]" id="note_note" style="overflow: hidden; overflow-wrap: break-word; resize: none; height: 140px;"></textarea>
```

This doesn't apply to sites like github, which also uses placeholder text, but I don't understand web coding well enough to understand what might be the difference.

```html
<textarea name="comment[body]" id="new_comment_field" placeholder="Leave a comment" aria-label="Comment body" class="form-control input-contrast comment-form-textarea js-comment-field js-paste-markdown js-task-list-field js-quick-submit js-size-to-fit js-session-resumable js-saved-reply-shortcut-comment-field" required=""></textarea>
```

If an obvious reason doesn't jump out at you, I'll check with a minimal rc.

**Comment by glacambre** [comment]
I can't reproduce this issue, could you try with the follwing firenvim_config?
`let g:firenvim_config = { 'localSettings': { '.*': { 'takeover': 'nonempty' } } }`

**Comment by BlueDrink9** [comment]
Can no longer reproduce. Thank you for you time. If it happens again I'll try the config.

**Comment by glacambre** [comment]
Great, thanks for letting me know! :)

**Comment by BlueDrink9** [comment]
I'm getting this happening on slack, with the following minimal configuration:

```
let g:firenvim_config = {
    \ 'localSettings': {
        \ '.*': {
            \ 'takeover': 'nonempty',
        \ },
    \ }
\ }
```

Do you use slack and mind seeing if you can repro?

This is the textbox div:

```
<div class="ql-editor ql-blank" data-gramm="false"
contenteditable="true"                                 id="undefined"
dir="auto"
role="textbox" tabindex="0"
data-team-id="T011PN0FFCM"
aria-label="Message
channel?p?"
aria-describedby="context_bar_text-f2a3a669"
aria-multiline="true"
aria-autocomplete="list"
aria-expanded="false"
aria-owns="chat_input_tab_ui"
spellcheck="true">
    <p>
        <br>
    </p>
</div>
    ```

Does `nonempty` check for all-whitespace chars, or if it contains a <br> in the paragraph?

**Comment by glacambre** [comment]
On 2020-05-03 06:49, BlueDrink9 wrote:
> Does `nonempty` check for all-whitespace chars, or if it contains a 
> <br> in the paragraph?

No, it just checks for an non-empty string. Trimming that string 
probably makes sense indeed.

**Comment by glacambre** [comment]
I've merged #545, which trims strings before checking if they're empty 
or not.

**Comment by BlueDrink9** [comment]
Thanks, I'll confirm at next release
